src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -296,6 +296,11 @@ private function registerBusMiddleware(ContainerBuilder $container, string $busI
                 $childDefinition = new ChildDefinition($messengerMiddlewareId);
                 $count = \count($definition->getArguments());
                 foreach (array_values($arguments ?? array()) as $key => $argument) {
+                    // Convert argument to references if needed.
+                    if (is_string($argument) && $argument[0] === '@') {
+                        $argument = new Reference(substr($argument, 1));
```

Comment:
Ping @ogizanagi, This is the best way I found to support references. Im not sure if this is needed if you configure `framework.messenger` with XML.